### PR TITLE
add targets to Makefile

### PR DESCRIPTION
**Pull request by leroxyl**
**Added:**

Makefile targets:
- lint
- build
- publish
- publish-branch
- clean
- tag-stable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,6 +27,19 @@ IMAGE_REPO := docker.io/ubirch/$(NAME)
 IMAGE_TAG := $(VERSION)
 
 GO_VERSION := 1.19
+LDFLAGS = -ldflags "-buildid= -s -w -X main.Version=$(VERSION) -X main.Revision=$(REVISION)"
+GO_BUILD = $(GO) build -tags="netgo" -trimpath $(LDFLAGS)
```
Comment (leroxyl) [inline Makefile:31]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -38,7 +53,26 @@ image:
...
 		--label="org.opencontainers.image.revision=$(REVISION)" .
 
-.PHONY: test
-test:
-	$(DOCKER) run -t --rm -v $(THISDIR)/main:/app -w /app golang:$(GO_VERSION) \
-	go test ./... -short
\ No newline at end of file
+.PHONY: build
+build:
+	CGO=1 GOOS=linux GOARCH=amd64 $(GO_BUILD) -o main .
```
Comment (leroxyl) [inline Makefile:58]:
```suggestion
build: image
```

Comment (leroxyl) [inline Makefile:58]:
```suggestion
build: image
```

Comment (leroxyl) [reply Makefile:58]: @ailox can we simply build via the dockerfile here instead?

Comment (ailox) [reply Makefile:58]: It is possible, but i think this should stay a seperate target.
* `make` or `make build` builds the executable binaries locally
* `make image` builds an image, ready for testing and uploading.

**Comment by ailox** [comment]
LGTM

**Pull request merged**